v3/rfcs/dx/20240620-simplify-subgraph-getting-started-and-federation.md

```diff
@@ -0,0 +1,47 @@
+# DX: simplify subgraphs for getting started and federation workflows
+
+Status: Review required
+
+## subgraph init command
+What this is: as a part of making it easier for people to work through the getting started guide, or to scaffold project files, allow the user to go from scratch to an API with only CLI commands.
+
+init-ing a supergraph or connector will create files for both local and cloud environments. This is about what happens for the subgraphs.
+
+```bash
+# creates a subgraph folder and subgraph.yaml file
+ddn subgraph init app
```

Comment:
Assuming this command is run in a DDN project directory (where we can locate `hasura.yaml`)? Then we create the `app` directory at the root of the project? It would not make a lot of sense if the directory in created in the CWD. We should also think about the variant for federation use case, where we init a standalone subgraph (thus creating `hasura.yaml`, `supergraph.local.yaml`, the `globals`  directory, engine, docker-compose files etc)

Reply:
Yes, let's keep the existing behaviour of putting next to hasura.yaml.

Let's also introduce a --dir flag to explicitly give the directory like we do with connectors.